### DISCUSSION TITLE
Roborock S7 dustbin removal/installation status

### DISCUSSION THREAD

**Discussion by manunited10** [rytilahti/python-miio#1419] [Ideas]
Hey guys.
This repo really rocks! well done.
I have a favour to ask for `Roborock S7`
I was wondering if it's possible to add a sensor to show if `dustbin` is removed or not? For sure the robot itself knows it and plays a sound confirmation.
The reason I'm asking for it is to add an automation for emptying the dustbin. I need to know when dustbin is removed/reinstalled to make decision.
Thanks guys.

**Comment by rytilahti** [comment]
Unfortunately I don't think that is possible. If you have rooted your device, maybe valetudo provides this information? https://github.com/Hypfer/Valetudo/issues/133 links to some 3rd party project, but considering that the issue is rather old, maybe valetudo has it nowadays built in?

**Reply by manunited10** [reply]
I usually do rooting/custom fw on my devices. But this one I just bought it - rather expensive. It's too soon to break it 😉
